### PR TITLE
[merp] Add breadcrumb for StackHash

### PR DESCRIPTION
**Pull request by alexischr**
Contributes to https://github.com/mono/mono/issues/18715

### CODE REVIEW

File: mono/utils/mono-state.c [hunk 1]
```diff
@@ -156,6 +156,12 @@ file_for_dump_reason_breadcrumb (const char *directory, const char *dump_reason,
 	g_snprintf (buff, sizeof_buff, "%s%scrash_reason_%s", directory, G_DIR_SEPARATOR_S, dump_reason);
 }
 
+static void
+file_for_hash_breadcrumb (const char *directory, MonoStackHash hashes, gchar *buff, size_t sizeof_buff)
+{
+	g_snprintf (buff, sizeof_buff, "%s%scrash_hash_%llu", directory, G_DIR_SEPARATOR_S, hashes.offset_rich_hash);
```
Comment (lambdageek) [inline mono/utils/mono-state.c]:
```suggestion
	g_snprintf (buff, sizeof_buff, "%s%scrash_hash_0x%" PRIx64 "", directory, G_DIR_SEPARATOR_S, (uint64_t)hashes.offset_rich_hash);
```

Should match what's emitted in `mono_merp_write_fingerprint_payload`

File: mono/metadata/threads.c [hunk 2]
```diff
@@ -6523,6 +6523,8 @@ summarizer_state_term (SummarizerGlobalState *state, gchar **out, gchar *mem, si
 		mono_get_eh_callbacks ()->mono_summarize_managed_stack (threads [i]);
 	}
 
+	mono_create_crash_hash_breadcrumb (controlling);
```
Comment (lambdageek) [inline mono/metadata/threads.c]:
```suggestion
	/* The value of the breadcrumb should match the "StackHash" value written by `mono_merp_write_fingerprint_payload` */
	mono_create_crash_hash_breadcrumb (controlling);
```

The control flow in crash reporting is very hard to follow.  We should add a test that the value that is written by `mono_merp_write_fingerprint_payload` is the same as the value of this breadcrumb.

**Comment by alexischr** [comment]
@monojenkins build failed

**Comment by alexischr** [comment]
@monojenkins build failed

**Comment by alexischr** [comment]
@monojenkins backport to 2020-02
@monojenkins backport to 2019-12

**Comment by alexischr** [comment]
@monojenkins backport to 2019-12
